numpy/ma/core.py

```diff
@@ -5284,6 +5311,10 @@ def argsort(self, axis=None, kind='quicksort', order=None,
 
         """
 
+        # 2017-04-11, Numpy 1.13.0, gh-8701: warn on axis default
+        if axis is np._NoValue:
+            axis = _deprecate_argsort_axis(self)
```

Comment:
Doing the test here makes it more obvious that this only affects default arguments